vmware-tanzu/nsx-operator
# Unify SecurityPolicy/Networkpolicy deletion process with GC and cleanup

In certain cases, SecurityPolicy or Networkpolicy deletion will fail because there is no VPC info found if VPC has been deleted before SP deletion.

Hence, this patch is to
1. Save NSX SecurityPolicy path after HAPI call into the store. This saved SecurityPolicy path will be used for deletion, GC or Cleanup to get VPC info.
2. Refactor SecurityPolicy creation/update and deletion process.
3. Unify SecurityPolicy normal deletion process with GC and cleanup,
which means it uses SecurityPolicy and NetworkPolicy UID to get NSX SecurityPolicy
from the store to avoid the rebuilding process. Because the rebuilding process also needs to
get VPC by VPC service, this step might fail.

The test done:
1. Create SecurityPolicy and Networkpolicy in VPC mode, and delete them successfully.
2. Create SecurityPolicy  in T1 mode, and delete them successfully.
3. Create SecurityPolicy and Networkpolicy in VPC mode, and delete them from K8s firstly, and do GC to see
created NSX SecurityPolicies are deleted by GC.
4. Create SecurityPolicy in T1 mode, and delete them from K8s firstly, and do GC to see
created NSX SecurityPolicies are deleted by GC.
